tests/components/flic/test_binary_sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Tests for Flic button integration."""
+from unittest import mock
+
+from homeassistant.setup import async_setup_component
+
+from tests.common import assert_setup_component
+
+
+class _MockFlicClient:
+    def __init__(self, button_addresses):
+        self.addresses = button_addresses
+        self.get_info_callback = None
+        self.scan_wizard = None
+
+    def close(self):
+        pass
+
+    def get_info(self, callback):
+        self.get_info_callback = callback
+        callback({"bd_addr_of_verified_buttons": self.addresses})
+
+    def handle_events(self):
+        pass
+
+    def add_scan_wizard(self, wizard):
+        assert isinstance(wizard, _MockScanWizard)
+        self.scan_wizard = wizard
+
+    def add_connection_channel(self, channel):
+        assert channel is None
+
+
+class _MockScanWizard:
+    on_completed = None
+
+
+async def test_button_uid(hass):
+    """Test UID assignment for Flic buttons."""
+    address_to_uid = {
+        "80:e4:da:78:6e:11": "flic_80e4da786e11",
+        "80:E4:DA:78:6E:12": "flic_80e4da786e12",  # Uppercase address should not change uid.
+    }
+
+    flic_client = _MockFlicClient(tuple(address_to_uid))
+
+    with mock.patch(
+        "homeassistant.components.flic.binary_sensor.FlicClient",
+        lambda _, __: flic_client,
+    ), mock.patch(
+        "homeassistant.components.flic.binary_sensor.FlicButton._create_channel",
```

Comment:
Please only patch the library, not our code.

Reply:
Done.

I've refactored `binary_sensor.py` to import the pyflic module rather than symbols in it to make it clearer from the test what is being patched (enables `mock.patch('pyflic.Foo', ..)` instead of `mock.patch('homeassistant.components.flic.binary_sensor.Foo', ..)`).